tls/s2n_config.c

```diff
@@ -808,6 +812,30 @@ int s2n_config_set_extension_data(struct s2n_config *config, s2n_tls_extension_t
     return 0;
 }
 
+int s2n_config_set_custom_critical_oids(struct s2n_config *config, const char *const *custom_critical_oids, uint32_t custom_oid_count)
+{
+    POSIX_ENSURE_REF(config);
+
+#ifndef S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID
+    POSIX_BAIL(S2N_ERR_INTERNAL_LIBCRYPTO_ERROR);
+#endif
+
+    if (config->custom_crit_oids != NULL) {
+        sk_ASN1_OBJECT_pop_free(config->custom_crit_oids, ASN1_OBJECT_free);
+    }
+    config->custom_crit_oids = sk_ASN1_OBJECT_new_null();
```

Comment:
The result of `sk_ASN1_OBJECT_new_null` should be checked:
```suggestion
    config->custom_crit_oids = sk_ASN1_OBJECT_new_null();
    POSIX_ENSURE_REF(config->custom_crit_oids);
```

And we should probably null `config->custom_crit_oids` after freeing it above, to be extra sure this call succeeded.